File: examples/Person.go
package main
import (
    "fmt"
)
// Propiedades
type Persona struct {
    nombre   string
    apellido string
    edad     int
}
// Constructor
func Persona(nombre, apellido string, edad int) *Persona {
    return &Persona{nombre, apellido, edad}
}
// Getters
func (p *Persona) NombreCompleto() string {
    return p.nombre + " " + p.apellido
}
func (p *Persona) Edad() int {
    return p.edad
}
// Setters
func (p *Persona) SetNombreCompleto(nombre, apellido string) {
    if len(nombre) >= 3 {
        p.nombre = nombre
    }
    if len(apellido) >= 3 {
        p.apellido = apellido
    }
}
func (p *Persona) SetEdad(edad int) {
    if edad >= 0 && edad <= 125 {
        p.edad = edad
    }
}
// Instancia
usuario := Persona("Cristian", "Racedo", 25)
fmt.Println(usuario.Edad()) // 25
usuario.SetEdad(33)
fmt.Println(usuario.Edad()) // 33

File: examples/Person.rb
class Persona
    # Propiedades
    attr_accessor 
        :nombre, 
        :apellido, 
        :edad
    # Metodos
    ## Constructor
    def initialize(nombre, apellido, edad)
        @nombre = nombre
        @apellido = apellido
        @edad = edad
    end
    ## Getters
    def nombre_completo
        "#{@nombre} #{@apellido}"
    end
    ## Setters
    def set_nombre_completo(nombre, apellido)
        @nombre = nombre if nombre.length >= 3
        @apellido = apellido if apellido.length >= 3
    end
    def set_edad(edad)
        @edad = edad if edad >= 0 && edad <= 125
    end
end
# Instancia
usuario = Persona.new("Cristian", "Racedo", 25)
puts usuario.edad # 25
usuario.set_edad(33)
puts usuario.edad # 33